public/app/features/explore/state/main.ts

```diff
@@ -50,90 +51,43 @@ export const maximizePaneAction = createAction<{
 export const evenPaneResizeAction = createAction('explore/evenPaneResizeAction');
 
 /**
- * Resets state for explore.
+ * Close the pane with the given id.
  */
-export const resetExploreAction = createAction('explore/resetExplore');
+type SplitCloseActionPayload = ExploreId;
+export const splitClose = createAction<SplitCloseActionPayload>('explore/splitClose');
 
-/**
- * Close the split view and save URL state.
- */
-export interface SplitCloseActionPayload {
-  itemId: ExploreId;
+export interface SetPaneStateActionPayload {
+  [itemId: string]: Partial<ExploreItemState>;
 }
-export const splitCloseAction = createAction<SplitCloseActionPayload>('explore/splitClose');
+export const setPaneState = createAction<SetPaneStateActionPayload>('explore/setPaneState');
 
-//
-// Action creators
-//
+export const clearPanes = createAction('explore/clearPanes');
 
 /**
- * Save local redux state back to the URL. Should be called when there is some change that should affect the URL.
- * Not all of the redux state is reflected in URL though.
- */
-export const stateSave = (options?: { replace?: boolean }): ThunkResult<void> => {
-  return (_, getState) => {
-    const { left, right } = getState().explore.panes;
-    const orgId = getState().user.orgId.toString();
-    const urlStates: { [index: string]: string | null } = { orgId };
-
-    urlStates.left = serializeStateToUrlParam(getUrlStateFromPaneState(left!));
-
-    if (right) {
-      urlStates.right = serializeStateToUrlParam(getUrlStateFromPaneState(right));
-    } else {
-      urlStates.right = null;
-    }
-
-    lastSavedUrl.right = urlStates.right;
-    lastSavedUrl.left = urlStates.left;
-
-    locationService.partial({ ...urlStates }, options?.replace);
-  };
-};
-
-// Store the url we saved last se we are not trying to update local state based on that.
-export const lastSavedUrl: UrlQueryMap = {};
-
-/**
- * Opens a new right split pane by navigating to appropriate URL. It either copies existing state of the left pane
- * or uses values from options arg. This does only navigation each pane is then responsible for initialization from
- * the URL.
+ * Opens a new split pane. It either copies existing state of the left pane
+ * or uses values from options arg.
+ *
+ * TODO: this can be improved by better inferring fallback values.
```

Comment:
Should we have a task in the backlog for it?

Reply:
will create tasks for each of the pending todo items once the PR is merged